domain/doc.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+// Package domain implements domain services.
+// Each domain service is responsible for providing APIs pertaining
+// to a logical domain. Domains may cross entity boundaries; the
+// partitioning is done based on packaging cohesive functional behaviour,
+// not individual entity concerns.
+//
+// # Package layout and conventions
+//
+// Each domain service package has several key artefacts:
+//   - the service providing public APIs called by API server facades.
+//   - the state providing functionality to read and write persistent data.
+//   - params structs which are used as arguments/results for service API calls.
+//   - arg structs which are used as arguments/results for state calls.
+//   - type structs which are used a sqlair in/out parameters.
+//
+// The layout of a domain package providing a service is as follows:
+//
+//		domain/foo/
+//		 |- entities.go [1]
```

Comment:
What differentiates this from types.go? There is on domain sub-package with such a file today.

Reply:
It was to contain domain structs moved down from core, as opposed to args structs used between the service and state layers. To avoid being too proscriptive, I've removed it from the first pass.